ABDataCollectionCore.js

```diff
@@ -1128,19 +1141,18 @@ module.exports = class ABDataCollectionCore extends ABMLClass {
 
                   // If the update item is current cursor, then should tell components to update.
                   let currData = this.getCursor();
-                  if (currData && currData.id == updatedVals.id) {
-                     this.emit("changeCursor", currData);
+                  if (currData?.id == updatedVals.id) {
+                     this.emit("cursorStale", currData);
```

Comment:
Code syntax looks good to me.

If we will remove this line (https://github.com/digi-serve/appbuilder_class_core/pull/210/files#diff-23eae8127a35e681261fee3393f168256b08a4b3807123f4666060de7359c2bcR1040), then this list should have correct relation data.
I curious why we need to reload data. I concern we will found the performance issue in the server on VPN again.
There are DCs who have the same the linked DC. They will load data at the same time.

Reply:
OK, you are asking about why we need to reload the data if the DC has just inserted the proper value in it's collection.

Just to be clear, the current DC doesn't reload it's data.  It has just updated it's value and is done.  

However, if that updated value was it's current cursor, then it does signal to any of it's Linked or Followed DCs that they need to reload their data. The change that was made might have an impact on what our linked DCs display.  That is the way we can guarantee we are displaying the correct data to the user.

So, you are right, in this case each DC that is linked to this one will perform an additional load of it's data.  (except for those that have .loadAll set, they just resort their data)

But this is the correct way to guarantee the proper data is displayed.

The same thing will happen if this DC changes it's cursor.  All the linked DCs will have to load() a new set of entries. 

If the value of the current cursor changes, we should treat it like it is a NEW value our Linked DCs need to respond to.

Reply:
Would be better if we can guarantee proper data is displayed by update without reload ?

Previously, When we have `stale`, We found a select cursor issue of child DCs also.
If they got load more next 20 data rows (by paging), then the user selects row item at 25 (by assume).
and the parent DC cursor will get update, then reload data of child DCs.
They have just only first [20 row items](https://github.com/digi-serve/ab_platform_web/blob/master/AppBuilder/platform/ABDataCollection.js#L149).
Those cursor of them will invalid.

Reply:
> Would be better if we can guarantee proper data is displayed by update without reload ?

I do agree with you.  That would be better.  I'd like to reduce the amount of data we are transferring back and forth between the browser and the server. So if we could just use the data from the updated record I would, but there is a problem with this data.

The problem we want to solve is the linked data being updated properly.  So, for example we have 2 DataCollections:
- `dcRoles`  : contains Fully Populated `Role` data, with `Users` that are connections.  The `User` data is the limited & unpopulated data we send back in our connected data field.
- `dcUsers`  : Linked to `dcRoles`, that contain fully populated `Users` that are linked to the currently selected `Role` in `dcRoles`.

Lets say we add a `User(Guy)` to the currently selected `Role(Builder)` in another part of the system.  

We get an update notifying us that `Role(Builder)` is updated.  The connected version of `Guy` is the limited/unpopulated set of data.

Now there are UI widgets connected to `dcUsers` that are expecting to work with the fully populated `User` value.  So we can't simply just take the value out of the `Role(Builder)->User(Guy)` because that is a limited value version.  

We have to make a request to pull that data from the server.

It is the same reason we have to pull that data from the server every time a cursor changes.

Reply:
> Previously, When we have `stale`, We found a select cursor issue of child DCs also. If they got load more next 20 data rows (by paging), then the user selects row item at 25 (by assume). and the parent DC cursor will get update, then reload data of child DCs. They have just only first [20 row items](https://github.com/digi-serve/ab_platform_web/blob/master/AppBuilder/platform/ABDataCollection.js#L149). Those cursor of them will invalid.

OK, that's a good insight. 

First thing that comes to my mind is: This seems like a complicated problem to solve.  Do we have some Tests in place to detect this?  If not, we really need to, because I'd hate to reintroduce that bug again and then have to debug it.

But that does bring up a point in doing software only resets of the data.  Maybe a software only initiation of reloading the data would get out of sync with the webix datacollection and their current paging.

So instead of me issuing a `cursorStale` event and responding to that, maybe we should have the Master Datacollection, simply:
```
let currentCursor = dcRoles->getCursor();
dcRoles->setCursor(null);
dcRoles->setCursor(currentCursor);
```

This would in effect sill perform the same task as the `cursorStale` event I just created, but also make sure that our webix datacollections are back in sync through the established (and trusty) `cursorChange` process. 

I think.

Should I make that change?  

I wish we had a test in place that would tell me if it was working ...

Reply:
Sure, I can create a test for you.